asdf/yamlutil.py

```diff
@@ -221,15 +223,42 @@ def custom_tree_to_tagged_tree(tree, ctx):
     directly representable in YAML, to a tree of basic data types,
     annotated with tags.
     """
-    def walker(node):
+    def _to_yaml_tree_tagged(converter, node):
+        node = converter.to_yaml_tree(node)
+
+        if isinstance(node, GeneratorType):
+            generator = node
+            node = next(generator)
+        else:
+            generator = None
+
+        node = _maybe_tag_node(converter, node)
+        yield node
+        if generator is not None:
+            yield from generator
+
+    def _maybe_tag_node(converter, node):
+        # If the subclass opted to select its own tag, don't
+        # overwrite its choice:
+        if isinstance(node, tagged.Tagged):
+            return node
+        else:
+            return tagged.tag_object(converter.tag, node, ctx=ctx)
+
+    def _walker(node):
+        converter = get_config().converter_index.from_type(type(node))
+        if converter is not None:
+            return _to_yaml_tree_tagged(converter, node)
+
```

Comment:
Does this coexist with the older TransformType? For example, won't it have to deal with tags not yet using the new Converter approach? This doesn't look like it handles that but I probably am missing where that is happening (e.g. those type classes presented as converters?)

Reply:
It does fall back to the Type classes when Converter support doesn't exist.  If object returned from `converter_index.from_type` is None, it proceeds to the original (unaltered) code.

Reply:
OK, I see how that is done now. Thanks